deepomatic/resources/task.py

```diff
@@ -40,30 +50,29 @@ def list(self, task_ids):
         assert(isinstance(task_ids, list))
         return super(Task, self).list(task_ids=task_ids)
 
-    def wait(self, timeout=60):
+    def wait(self, timeout=60, wait_exp_multiplier=0.1, wait_exp_max=1.0):
         """
         Wait until task is completed. Expires after 'timeout' seconds.
         """
-        self._wait_result(timeout)
-        return self
-
-    def _wait_result(self, timeout):
-        start_time = time.time()
-        last_time = start_time
-        sleep_time = 0.2
-        while self['status'] == "pending":
-            # Check for timeout
-            if time.time() > start_time + timeout:
+        try:
+            retryer = Retrying(wait=wait_random_exponential(multiplier=wait_exp_multiplier, max=wait_exp_max),
+                               stop=stop_after_delay(timeout),
+                               retry=retry_if_result(is_pending_status),
+                               before=before_log(logger, logging.DEBUG),
+                               after=after_log(logger, logging.DEBUG))
+            retryer(self._refresh_status)
+        except RetryError:
+            if self['status'] == 'pending':
                 raise TaskTimeout(self.data())
+            elif self['status'] == 'error':
```

Comment:
What happens if the status is 'error' ? Is RetryError thrown ?

Reply:
Indeed, I made a mistake, the retry is done only if status is pending, I just fixed it.